File: lib/models/location_model.dart
class MapLocation {
  int? _totalSize;
  int? _typeId;
  late List<LocationModel> _locations;
  List<LocationModel> get locations => _locations;

  MapLocation(
      {required totalSize,
      required typeId,
      required offset,
      required products}) {
    this._totalSize = totalSize;
    this._typeId = typeId;
    this._locations = products;
  }

  MapLocation.fromJson(Map<String, dynamic> json) {
    _totalSize = json['total_size'];
    _typeId = json['type_id'];
    if (json['locations'] != null) {
      _locations = <LocationModel>[];
      json['locations'].forEach((v) {
        _locations.add(LocationModel.fromJson(v));
      });
    }
  }
}

class LocationModel {
  int? id;
  String? addressType;
  String? address;
  double? latitude;
  double? longitude;
  int? typeId;
  String? createdAt;
  String? updatedAt;

  LocationModel(
      {this.id,
      this.addressType,
      this.address,
      this.latitude,
      this.longitude,
      this.createdAt,
      this.updatedAt,
      this.typeId});

  LocationModel.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    addressType = json['address_type'];
    address = json['address'];
    latitude = json['latitude'];
    longitude = json['longitude'];
    typeId = json['type_id'];
    createdAt = json['created_at'];
    updatedAt = json['updated_at'];
  }

  Map<String, dynamic> toJson() {
    return {
      "id": this.id,
      "address_type": this.addressType,
      "address": this.address,
      "latitude": this.latitude,
      "longitude": this.longitude,
      "type_id": this.typeId,
      "createdAt": this.createdAt,
      "updatedAt": this.updatedAt,
    };
  }
}

File: RIVA Terminal/Models/LocationModel.cs
using Microsoft.AspNetCore.Components;
using System.ComponentModel.DataAnnotations;

namespace RIVA.Models
{
    public class LocationModel : ComponentBase
    {
        public int id { get; set; }

        //[RegularExpression("[0-9]{6}", ErrorMessage = "Client No must be a 6 digit number")]
        [Display(Name = "Address Type")]
        [DataType(DataType.Text)]
        public string address_type { get; set; }

        [Display(Name = "Address")]
        [DataType(DataType.Text)]
        public string address { get; set; }

        [RegularExpression("[0-9.]{1,}$", ErrorMessage = "Latitude must be a float value")]
        [Display(Name = "Latitude")]
        [DataType(DataType.Text)]
        public string latitudeTxt { get; set; }
        public double latitude { get; set; }

        [RegularExpression("[0-9.]{1,}$", ErrorMessage = "Longitude must be a float value")]
        [Display(Name = "Longitude")]
        [DataType(DataType.Text)]
        public string longitudeTxt { get; set; }
        public double longitude { get; set; }

        [Display(Name = "Type Id")]
        [DataType(DataType.Text)]
        public string type_idTxt { get; set; }
        public int type_id { get; set; }

        public string? created_at { get; set; }
        public string? updated_at { get; set; }
    }

    public class LocationRoot
    {
        public int total_size { get; set; }
        public int type_id { get; set; }
        public List<LocationModel> locations { get; set; }
    }
}
